airflow/models.py

```diff
@@ -1956,6 +1956,8 @@ class derived from this one results in the creation of a task object,
     template_ext = []
     # Defines the extra buttons to display in the task instance model view
     extra_links = []
+    # Defines which function to call per button
+    extra_link_functions = {}
```

Comment:
I think we should go all the way, and replace `extra_links` with `extra_link_functions`

Reply:
Looking at the js code, I think we still need to provide a list of buttons for UI to render. I can make `extra_links` a property that returns the keys (button names) from `extra_link_functions`.